streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -219,11 +219,7 @@ private boolean waitOnState(final State targetState, final long waitMs) {
             long elapsedMs = 0L;
             while (state != targetState) {
                 if (waitMs == 0) {
-                    try {
-                        stateLock.wait();
```

Comment:
Why do we remove this? Would it be better to call it with `stateLock.wait(0L);` ? (or better merge with case below and update condition to `if (waitMs >= elapsedMs)` ?